src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -1833,6 +1840,7 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
     //   swapchain size. This helper will call _DoResize with the current size
     //   of the swapchain, accounting for scaling due to DPI.
     // - Note that a DPI change will also trigger a font size change, and will call into here.
+    // - The write lock should be held when calling this method, we might be changing the buffer size in _DoResize.

```

Comment:
Is there any way we can check if a lock is currently held? Like introduce some sort of `assert(terminal->IsLocked())`?

Reply:
hmm, I don't think you can ask if the mutex is held bc of the whole "oh it's not held @ time = 0? let me lock then! (but then somebody else swipes it away @ time = 0.001)", but there is a `std::shared_mutex::try_lock` that would return false if it can't lock otherwise it returns true and locks? 🤔